### ISSUE TITLE
Failed to execute "'python' '$PWD/main.py'" if empty string in wildignore

### ISSUE THREAD

**Issue by thomwiggers** [vim/vim#710]
When I run `vim editname`, I get the following:

```
Failed to execute "'python' '/home/thom/path/main.py' '/home/thom/path/editname'". Exit code: 2
Message: 
['python: can''t open file ''/home/thom/path/main.py'': [Errno 2] No such file or directory']
```

every time I open a file. I've tracked it down to the following:

In my `.vimrc` I had this line:

```
set wildignore+=foo,,bar
```

Note the `,,`. If I normalise this to `,` the error goes away.

This error didn't occur in the 1529 version.

**Comment by vim-ml** [comment]
Am 2016-03-23 15:12, schrieb Thom Wiggers:

> When I run vim editname, I get the following:
> 
> Failed to execute "'python' '/home/thom/path/main.py'
> '/home/thom/path/editname'". Exit code: 2
> Message:
> ['python: can''t open file
> ''/home/thom/git/tueviff/TUeVIFFie/main.py'': [Errno 2] No such file
> or directory']
> 
> every time I open a file. I've tracked it down to the following:
> 
> In my .vimrc I had this line:
> 
> set wildignore+=foo,,bar
> 
> Note the ,,. If I normalise this to , the error goes away.

What version is this?

Best,
Christian

**Comment by thomwiggers** [comment]
I've noticed it happening in both 1584 and 1639 (Arch linux' gvim package and gvim-git from AUR) .

**Comment by mnebuerquo** [comment]
I am experiencing this issue as well. I get the exact same error message as written above with the $PWD of my file.

It happens when I have this line in my .vimrc:

```
set wildignore+=*/tmp/*,*.so,*.swp,*.zip,node_modules
```

This is my current version of vim:

```
VIM - Vi IMproved 7.3 (2010 Aug 15, compiled Oct 28 2015 19:46:19)
Compiled by root@apple.com
Normal version without GUI.  Features included (+) or not (-):
-arabic +autocmd -balloon_eval -browse +builtin_terms +byte_offset +cindent
-clientserver -clipboard +cmdline_compl +cmdline_hist +cmdline_info +comments
-conceal +cryptv +cscope +cursorbind +cursorshape +dialog_con +diff +digraphs
-dnd -ebcdic -emacs_tags +eval +ex_extra +extra_search -farsi +file_in_path
+find_in_path +float +folding -footer +fork() -gettext -hangul_input +iconv
+insert_expand +jumplist -keymap -langmap +libcall +linebreak +lispindent
+listcmds +localmap -lua +menu +mksession +modify_fname +mouse -mouseshape
-mouse_dec -mouse_gpm -mouse_jsbterm -mouse_netterm -mouse_sysmouse
+mouse_xterm +multi_byte +multi_lang -mzscheme +netbeans_intg -osfiletype
+path_extra -perl +persistent_undo +postscript +printer -profile +python/dyn
-python3 +quickfix +reltime -rightleft +ruby/dyn +scrollbind +signs
+smartindent -sniff +startuptime +statusline -sun_workshop +syntax +tag_binary
+tag_old_static -tag_any_white -tcl +terminfo +termresponse +textobjects +title
 -toolbar +user_commands +vertsplit +virtualedit +visual +visualextra +viminfo
+vreplace +wildignore +wildmenu +windows +writebackup -X11 -xfontset -xim -xsmp
 -xterm_clipboard -xterm_save
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
      user exrc file: "$HOME/.exrc"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -D_FORTIFY_SOURCE=0 -Iproto -DHAVE_CONFIG_H -arch i386 -arch x86_64 -g -Os -pipe
Linking: gcc -arch i386 -arch x86_64 -o vim -lncurses
```

**Comment by thomwiggers** [comment]
@mnebuerquo It may be interesting to post your full `wildignore` setting, you can get it from Vim using the command `:set wildignore?`.

**Comment by mnebuerquo** [comment]
@thomwiggers , actually, that's the whole thing. I only used the += in my vimrc in case I was going to add more to it.

**Comment by thomwiggers** [comment]
Oh, and relevant to the original report: I'm using a bunch of plugins. My vimrc is [listed here](https://github.com/thomwiggers/vcsh-vim/blob/master/.vimrc). I still think this is a vim issue though, since the plugins weren't updated between the vim version upgrade that caused the bug to appear.

**Comment by mnebuerquo** [comment]
I also use a bunch of plugins. They're installed using Neobundle.

[My vimrc references several other vim files in a separate rc directory in my dotfiles repo.](https://github.com/mnebuerquo/dotfiles/tree/master/vim)

My wildignore setting is in vim/rc/ag.vim in the above repo.

I don't know if this is vim or a plugin, but google search led me to this issue.

**Comment by mnebuerquo** [comment]
I am seeing this on another vim install, this time on Ubuntu instead of a Mac. This one uses the same dotfiles repo as in my previous comment.
Here is my `:set wildignore?` output:

```
wildignore=*/tmp/*,*.so,*.swp,*.zip,node_modules
```

Here is the output from `vim --version`:

```
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Aug 15 2014 04:43:01)
Included patches: 1-273
Modified by pkg-vim-maintainers@lists.alioth.debian.org
Compiled by buildd@
Huge version with GTK2 GUI.  Features included (+) or not (-):
+acl             +farsi           +mouse_netterm   +syntax
+arabic          +file_in_path    +mouse_sgr       +tag_binary
+autocmd         +find_in_path    -mouse_sysmouse  +tag_old_static
+balloon_eval    +float           +mouse_urxvt     -tag_any_white
+browse          +folding         +mouse_xterm     +tcl
++builtin_terms  -footer          +multi_byte      +terminfo
+byte_offset     +fork()          +multi_lang      +termresponse
+cindent         +gettext         -mzscheme        +textobjects
+clientserver    -hangul_input    +netbeans_intg   +title
+clipboard       +iconv           +path_extra      +toolbar
+cmdline_compl   +insert_expand   +perl            +user_commands
+cmdline_hist    +jumplist        +persistent_undo +vertsplit
+cmdline_info    +keymap          +postscript      +virtualedit
+comments        +langmap         +printer         +visual
+conceal         +libcall         +profile         +visualextra
+cryptv          +linebreak       +python          +viminfo
+cscope          +lispindent      -python3         +vreplace
+cursorbind      +listcmds        +quickfix        +wildignore
+cursorshape     +localmap        +reltime         +wildmenu
+dialog_con_gui  +lua             +rightleft       +windows
+diff            +menu            +ruby            +writebackup
+digraphs        +mksession       +scrollbind      +X11
+dnd             +modify_fname    +signs           -xfontset
-ebcdic          +mouse           +smartindent     +xim
+emacs_tags      +mouseshape      -sniff           +xsmp_interact
+eval            +mouse_dec       +startuptime     +xterm_clipboard
+ex_extra        +mouse_gpm       +statusline      -xterm_save
+extra_search    -mouse_jsbterm   -sun_workshop    +xpm
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "$VIM/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread -I/usr/include/gtk-2.0 -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/pango-1.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/freetype2 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/libpng12 -I/usr/include/harfbuzz     -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1     -I/usr/include/tcl8.6  -D_REENTRANT=1  -D_THREAD_SAFE=1  -D_LARGEFILE64_SOURCE=1  
Linking: gcc   -L. -Wl,-Bsymbolic-functions -Wl,-z,relro -L/build/buildd/ruby1.9.1-1.9.3.484/debian/lib -rdynamic -Wl,-export-dynamic -Wl,-E  -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--as-needed -o vim   -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgio-2.0 -lpangoft2-1.0 -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lcairo -lpango-1.0 -lfontconfig -lgobject-2.0 -lglib-2.0 -lfreetype   -lSM -lICE -lXpm -lXt -lX11 -lXdmcp -lSM -lICE  -lm -ltinfo -lnsl  -lselinux  -lacl -lattr -lgpm -ldl  -L/usr/lib -llua5.2 -Wl,-E  -fstack-protector -L/usr/local/lib  -L/usr/lib/perl/5.18/CORE -lperl -ldl -lm -lpthread -lcrypt -L/usr/lib/python2.7/config-x86_64-linux-gnu -lpython2.7 -lpthread -ldl -lutil -lm -Xlinker -export-dynamic -Wl,-O1 -Wl,-Bsymbolic-functions  -L/usr/lib/x86_64-linux-gnu -ltcl8.6 -ldl -lz -lpthread -lieee -lm -lruby-1.9.1 -lpthread -lrt -ldl -lcrypt -lm  -L/usr/lib
```

**Comment by mnebuerquo** [comment]
I think I've made a breakthrough, and I'd like for someone to confirm it. I have removed and re-added all of my plugins, to see if any affect the issue. When all were removed, opening a file did not cause this error. I incrementally added them back in to find any that triggered it.

[editorconfig/editorconfig-vim](https://github.com/editorconfig/editorconfig-vim) seems to be the culprit.

Just to be clear, I think the problem went away when I removed the `wildignore` setting on the Mac. On Ubuntu that did not fix it, but removing the `editorconfig-vim` plugin did fix it. I will be able to try both solutions again tonight on the Mac to double-check. It may be there is an interaction between the two.

@thomwiggers , are you using this plugin? If yes, and you delete this plugin from your bundles directory, does the problem go away?

**Comment by thomwiggers** [comment]
I am not using that plugin. Maybe it was the only plugin in your setup that was using Python? I do suspect some relation to the python subsystems in Vim.

**Comment by mnebuerquo** [comment]
I have confirmed that this error goes away when I do either fix on the Mac. I can remove the `wildignore` setting, or I can omit the `editorconfig-vim` plugin I mentioned above. 

If both the `wildignore` setting and the `editorconfig-vim` plugin are added into my configuration at once, then I get the python error every time I open a file.

**Comment by misterfish** [comment]
For what it's worth: I was unable to reproduce this bug using either the arch@1639 revision @thomwiggers mentioned with his dot files and plugins, or compiling from source using @mnebuerquo's exact configuration and his dot files and plugins.

@thomwiggers Perhaps you can disable all your plugins and re-enable them one by one like @mnebuerquo did?

**Comment by thomwiggers** [comment]
Oh, huh, it turns out I _do_ have the editorconfig plugin, and if I disable it, it causes this issue to disappear.

**Comment by mnebuerquo** [comment]
So the only weird part of this is on my Mac which suggests that the wildignore has an additional effect. Maybe it is some form of interaction between path settings? I'll retest that later tonight to be sure.

In the event that editorconfig is the root cause, I have found a recent commit there which I think is related: 
https://github.com/editorconfig/editorconfig-vim/commit/75fe5721c456538c1ef2ded60b29f080feaac577
I'm not really sure how this works, or how the wildignore could be related, but that change occurred between the time I first installed it with no errors and the time I reinstalled and started having the errors.

**Comment by misterfish** [comment]
@mnebuerquo Well spotted. Now I can reproduce it by the way. And I believe you can take your whole rc/ directory out of the equation -- the bug is there without it as well.

You have `set runtimepath+=~/.vim/bundle/*/` in your vimrc.

After fnameescape() in the plugin the star gets escaped, and `l:python_core_files_dir` is empty.

Technically a wildcard is allowed, so it seems to be a bug in the plugin and in any case not to be a vim issue.

**Comment by mnebuerquo** [comment]
@misterfish, thanks for figuring this out. I'll post an issue on the plugin and link to this one for reference.

**Comment by misterfish** [comment]
No worries.

**Comment by mnebuerquo** [comment]
Opened issue in the editorconfig repo:
https://github.com/editorconfig/editorconfig-vim/issues/63

**Comment by thomwiggers** [comment]
@chrisbra so why did you close this? This issue only cropped up after vim was updated. I did not update editorconfig in the intermediate. I confirmed that this issue did not occur with the previous version of Vim as reported in the initial report, while I was using the exact same Vim configuration. So why is this a problem with just editorconfig, and not Vim? There is a difference in behaviour between Vim versions...
